### PR TITLE
 Implement multi root workspace folder changes

### PR DESCRIPTION
**Pull request by Arcticae**
Moved from https://github.com/starkware-libs/cairo/pull/6715

### CODE REVIEW

**Review by mkaput — APPROVED** [review]

**Comment by mkaput** [comment]
@Draggu please copy your review from original PR

**Review by mkaput — APPROVED** [review]

File: src/extension.ts [hunk 1]
```diff
@@ -1,27 +1,44 @@
...
+    // Notify the server when the client configuration changes.
+    // CairoLS pulls configuration properties it is interested in by itself, so it
+    // is unnecessary to attach any details in the notification payload.
+    ctx.extension.subscriptions.push(
+      vscode.workspace.onDidChangeConfiguration(
+        async () => {
+          await extensionManager
+            .getClient()
+            ?.sendNotification(lc.DidChangeConfigurationNotification.type, {
+              settings: "",
```
Comment (Draggu) [inline src/extension.ts:22]: Can we use `null` here instead of string?
```suggestion
              settings: null,
```

**Review by Draggu — CHANGES_REQUESTED** [review]

File: src/extension.ts (hunk 1 above)
Comment (Arcticae) [reply src/extension.ts:22]: It's out of scope because it's only moved around

Comment (Draggu) [reply src/extension.ts:22]: ok

**Review by Draggu — APPROVED** [review]

**Pull request merged**